src/mongo/db/pipeline/expression_test.cpp

```diff
@@ -2864,6 +2864,41 @@ TEST(ExpressionObjectOptimizations, OptimizingAnObjectShouldOptimizeSubExpressio
     ASSERT_VALUE_EQ(expConstant->evaluate(Document()), Value(3));
```

Comment:
I think you can adapt this test to suit your needs.

Instead of asserting that the inner thing is a constant, assert that the 'optimized' variable itself is an ExpressionConstant, and that it has value `Value(Document{{"a", 3}})`